cyclr.js

```diff
@@ -328,3 +334,52 @@ function setRecordFieldValue(record, fieldName, fieldValue) {
     var transformed = getRecordFieldValue(record, fieldName, fieldValue);
     record.setFieldValue(fieldName, transformed);
 }
+
+function setSubscriptions(record, datain) {
+    const subscriptionsFieldName = 'subscriptions';
+    var count = record.getLineItemCount(subscriptionsFieldName);
+    if (count == 0)
+        return; // No subscriptions to set.
+
+    // Get the subscription names/internal ids.
+    var subscriptionNameMap = JSON.parse(JSON.stringify(record))[subscriptionsFieldName].map(function(i) { return i.subscription });
+
+    for (var i = 1; i <= count; i++) {
+        var subscriptionId = record.getLineItemValue(subscriptionsFieldName, 'subscription', i);
+        
+        // Find the subscription object.
+        var subscription = null;
+        for (var c = 0; c < subscriptionNameMap.length; c++) {
+            if (subscriptionNameMap[c].internalid === subscriptionId) {
+                subscription = subscriptionNameMap[c];
+                break;
+            }
+        }
+
+        if (subscription == null)
+            continue;
+            
+        // Find the datain that matches the subscription.
+        var datainSubscription = null;
+        for (var c = 0; c < datain.subscriptions.length; c++) {
+            // Find the subscription by ID or name, if ID provided name will be ignored.
+            if (datain.subscriptions[c].id) {
+                if (datain.subscriptions[c].id === subscription.internalid)
+                    datainSubscription = datain.subscriptions[c];
+                    break;

```

Comment:
This will always exit the loop after the first ID match. Change it to
```javascript
            // Find the subscription by ID or name, if ID provided name will be ignored.
            if (datain.subscriptions[c].id === subscription.internalid) {
                datainSubscription = datain.subscriptions[c];
                break;
            }
            else if (datain.subscriptions[c].name === subscription.name) {
                datainSubscription = datain.subscriptions[c];
                break;
            }
```

Also this doesn't match the fields in our connector: https://my.cyclr.com/admin/connectors/192/releases/11323/methods/207929/edit#request
 Subscription ID is  [subscriptions].subscription.internalid 
 Subscription Name is missing in our connector, but it should be [subscriptions].subscription.name

(Needs testing)
```javascript
            // Find the subscription by ID or name, if ID provided name will be ignored.
            if (datain.subscriptions[c].subscription.internalid === subscription.internalid) {
                datainSubscription = datain.subscriptions[c];
                break;
            }
            else if (datain.subscriptions[c].subscription.name === subscription.name) {
                datainSubscription = datain.subscriptions[c];
                break;
            }
```